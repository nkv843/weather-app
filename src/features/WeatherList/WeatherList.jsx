import React from 'react';
import PropTypes from 'prop-types';
import WeatherItem from './components/WeatherItem';
import classNames from './WeatherList.module.css';

const WeatherList = ({ weathers }) => (
  <div className={classNames.list}>
    {weathers?.map((weather) => (
      <WeatherItem
        weather={weather}
        key={weather.id}
      />
    ))}
  </div>
);

WeatherList.propTypes = {
  weathers: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default WeatherList;
