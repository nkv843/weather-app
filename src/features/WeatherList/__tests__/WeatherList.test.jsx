import { render } from '@testing-library/react';
import React from 'react';
import WeatherList from '../WeatherList';
import weathers from '../__fixtures__/fixtures';

describe('WeatherList', () => {
  const { baseElement } = render(<WeatherList weathers={weathers} />);
  it('should match snapshot', () => {
    expect(baseElement).toMatchSnapshot();
  });
});
