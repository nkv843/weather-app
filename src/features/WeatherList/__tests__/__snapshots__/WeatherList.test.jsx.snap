// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherList should match snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="item"
      >
        <div
          class="item__content"
        >
          <div
            class="item__current"
          >
            <div
              class="item__header"
            >
              <div
                class="item__location"
              >
                Chișinau, Moldova
              </div>
              <div
                class="item__time"
              >
                17:02
              </div>
              <div
                class="item__temp"
              >
                8
                 °C
              </div>
              <img
                alt="weather"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                style="height: 100%;"
              />
              <div
                class="item__descr"
              >
                Cloudy
              </div>
            </div>
            <div
              class="item__dailyForecast daylyForecast"
            >
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  13:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  6
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  16:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  8
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  19:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  7
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  22:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  7
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  1:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/partlyCloudyNight.52162ae322e5349c343db37a8ef1d1bf.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  6
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  4:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/partlyCloudyNight.52162ae322e5349c343db37a8ef1d1bf.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  5
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  7:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  5
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  10:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  5
                   
                  °C
                </p>
              </div>
            </div>
            <div
              class="item__currentCondition"
            >
              <div
                class="item__wind"
              >
                <div
                  class="item__windSpeed"
                >
                  light, 0.3-3.4 m/s
                </div>
                <div
                  class="item__windDirection"
                >
                  W
                </div>
              </div>
              <div
                class="item__cloudiness"
              >
                100%
              </div>
              <div
                class="item__humidity"
              >
                60%
              </div>
              <div
                class="item__pressure"
              >
                760
                 
                mm Hg
              </div>
            </div>
          </div>
          <div
            class="verticalDivider"
          />
          <div
            class="item__weeklyForecast weeklyForecast"
          >
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                17 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/lightRain.0352bc773c5fea9a0db727ea8a608c38.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                6
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                18 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                6
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                19 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/lightRain.0352bc773c5fea9a0db727ea8a608c38.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                4
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                20 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -1
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                21 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -1
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                22 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/snow.9bb03875f08a8a3a46aab97c632982f7.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -1
                 
                °C
              </p>
            </div>
          </div>
        </div>
        <button
          class="item__addToFav"
          type="button"
        >
          <svg
            class="item__addToFavIcon"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 480a16 16 0 01-10.63-4L256 357.41 122.63 476A16 16 0 0196 464V96a64.07 64.07 0 0164-64h192a64.07 64.07 0 0164 64v368a16 16 0 01-16 16z"
            />
          </svg>
        </button>
      </div>
      <div
        class="item"
      >
        <div
          class="item__content"
        >
          <div
            class="item__current"
          >
            <div
              class="item__header"
            >
              <div
                class="item__location"
              >
                Moscow, Russia
              </div>
              <div
                class="item__time"
              >
                18:02
              </div>
              <div
                class="item__temp"
              >
                -1
                 °C
              </div>
              <img
                alt="weather"
                src="/static/media/clearDay.1d6771ec3e12034e10b1f8a99c08aca5.svg"
                style="height: 100%;"
              />
              <div
                class="item__descr"
              >
                Clear
              </div>
            </div>
            <div
              class="item__dailyForecast daylyForecast"
            >
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  14:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/clearDay.1d6771ec3e12034e10b1f8a99c08aca5.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -2
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  17:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/clearDay.1d6771ec3e12034e10b1f8a99c08aca5.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -1
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  20:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/clearNight.d7d8442be60c1ed9ab9e759c83c7b763.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -4
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  23:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/clearNight.d7d8442be60c1ed9ab9e759c83c7b763.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -4
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  2:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/clearNight.d7d8442be60c1ed9ab9e759c83c7b763.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -5
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  5:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/partlyCloudyNight.52162ae322e5349c343db37a8ef1d1bf.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -4
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  8:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -4
                   
                  °C
                </p>
              </div>
              <div
                class="forecast__item"
              >
                <p
                  class="forecast__itemTitle"
                >
                  11:00
                </p>
                <img
                  alt="weather"
                  class="forecast__itemIcon"
                  src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
                />
                <p
                  class="forecast__itemTemp"
                >
                  -4
                   
                  °C
                </p>
              </div>
            </div>
            <div
              class="item__currentCondition"
            >
              <div
                class="item__wind"
              >
                <div
                  class="item__windSpeed"
                >
                  light, 0.3-3.4 m/s
                </div>
                <div
                  class="item__windDirection"
                >
                  NE
                </div>
              </div>
              <div
                class="item__cloudiness"
              >
                10%
              </div>
              <div
                class="item__humidity"
              >
                63%
              </div>
              <div
                class="item__pressure"
              >
                760
                 
                mm Hg
              </div>
            </div>
          </div>
          <div
            class="verticalDivider"
          />
          <div
            class="item__weeklyForecast weeklyForecast"
          >
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                17 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -3
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                18 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/occasionalSnowNight.b3a56a624440f5ee367a3f29f2f4e482.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -3
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                19 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/snow.9bb03875f08a8a3a46aab97c632982f7.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -2
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                20 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -3
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                21 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -4
                 
                °C
              </p>
            </div>
            <div
              class="forecast__item"
            >
              <p
                class="forecast__itemTitle"
              >
                22 Nov
              </p>
              <img
                alt="weather"
                class="forecast__itemIcon"
                src="/static/media/cloudy.4de8ef8e0397e74192a471da49101ac0.svg"
              />
              <p
                class="forecast__itemTemp"
              >
                -5
                 
                °C
              </p>
            </div>
          </div>
        </div>
        <button
          class="item__addToFav"
          type="button"
        >
          <svg
            class="item__addToFavIcon"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 480a16 16 0 01-10.63-4L256 357.41 122.63 476A16 16 0 0196 464V96a64.07 64.07 0 0164-64h192a64.07 64.07 0 0164 64v368a16 16 0 01-16 16z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
